src/dune_engine/action_exec.ml

```diff
@@ -264,14 +278,15 @@ let exec_run_dynamic_client ~display ~ectx ~eenv prog args =
            that is incompatible with this version of dune."
           prog_name
       ]
-  | Ok Done -> Done
+  | Ok Done -> {done_or_more_deps = Done; needed_deps = Dep.Set.empty}
```

Comment:
Suggest define `return : ?needed_deps:Dep.Set.t -> done_or_more_deps -> action_res` and use that instead of these record literals.